src/core/useSpriteLoader.tsx

```diff
@@ -1,19 +1,66 @@
+/* eslint react-hooks/exhaustive-deps: 1 */
 import { Texture, TextureLoader } from 'three'
-import { useLoader, useThree } from '@react-three/fiber'
+import { useLoader, useThree, Vector3 } from '@react-three/fiber'
 import { useState } from 'react'
 import * as React from 'react'
 import * as THREE from 'three'
 
+type Size = {
+  w: number
+  h: number
+}
+
+type FrameData = {
+  frame: {
+    x: number
+    y: number
+    w: number
+    h: number
+  }
+  rotated: boolean
+  trimmed: boolean
+  spriteSourceSize: {
+    x: number
+    y: number
+    w: number
+    h: number
+  }
+  sourceSize: Size
+}
+
+type MetaData = {
+  version: string
+  size: {
+    w: number
+    h: number
+  }
+  rows: number
+  columns: number
+  frameWidth: number
+  frameHeight: number
+  scale: string
+}
+
+type SpriteData = {
+  frames: Record<string, FrameData[]> | FrameData[]
+  meta: MetaData
+}
+
+type SpriteMetaDimension = {
+  row: number
+  col: number
+}

```

Comment:
can't reuse any type from SpriteAnimator?

Reply:
should I export from spriteanimator? I thought it would be a good idea to have them not rely on each other, after all useSpriteLoader can be used as a standalone hook to load up textures from a sprite, not necessarily used with SpriteAnimator. 
But I do understand that we have duplicate types now and that could be an issue, perhaps if SpriteAnimator could pull types from useSpriteLoader since the latter is more generic? What do you think?

Reply:
agreed, I'll see to it, cheers!

Reply:
I've added these changes, SpriteAnimator pulls types when possible from useSpriteLoader